Dockerfile

```diff
@@ -10,15 +10,32 @@ WORKDIR $HOME
 
 # copy over install_files/ for use in playbooks
 ADD install_files $HOME/install_files
+RUN apt update && apt install -y sudo
+
+# fix for .gnupg/ permissions when building custom images
+RUN apt install -y qgis
```

Comment:
Any reason we're installing qgis (as well as postgresql, postgis, pg admin, & python packages) in the Dockerfile as opposed to in the ansible playbook?

Reply:
Ansible was a massive bottleneck for us the way we were previously using it. The way we're using it now, Dockerfiles allow us to cache each installation step in a layer whereas previously the entire Ansible playbook ran in a single layer. This meant that for any small change we made to the playbook, Docker would have to rebuild that entire layer. By separating this out we're increasing the number of layers but we're saving a lot of time as far as building the container is concerned because each of those layers can be cached. IMO we should never use Ansible for installation, and only rely on it for more complex configurations that would be too difficult to do with a Dockerfile. 

tl;dr Dockerfiles are much better as far as installing software packages versus Ansible for our needs

Reply:
Fair points; let's use Dockerfile for installations & Ansible playbooks for managing configuration